File: Api/TeslaRent.Application/Cars/Models/CarDto.cs
using TeslaRent.Application.Common.AutoMapper;
using TeslaRent.Domain.Entities;

namespace TeslaRent.Application.Cars.Models;

public class CarDto : IMapFrom<Car>
{
    public int Id { get; set; }
    public string Vin { get; set; }
    public CarModelDto Model { get; set; }
    public double Mileage { get; set; }
    public DateTime ProductionDate { get; set; }

    public ICollection<CarStatusDto> CarStatusHistory { get; set; }
    public ICollection<CarLocationDto> CarLocationHistory { get; set; }
}

File: web/src/api/models/Cars/CarDto.ts
import {CarModelDto} from "./CarModelDto";
import {CarStatusDto} from "./CarStatusDto";
import {CarLocationDto} from "./CarLocationDto";

export interface CarDto {
    id: number;
    vin: string;
    model: CarModelDto;
    mileage: number;
    productionDate: string;
    carStatusHistory: CarStatusDto[];
    carLocationHistory: CarLocationDto[];
}
